src/seraphis_impl/enote_store_utils.cpp

```diff
@@ -111,10 +136,7 @@ static boost::multiprecision::uint128_t get_balance_intermediate_legacy(
                         "get balance intermediate legacy: tracked legacy duplicates has an entry that "
                         "doesn't line up 1:1 with the legacy intermediate map even though it should (bug).");
 
-                    return legacy_intermediate_records
-                        .at(identifier)
-                        .origin_context
-                        .origin_status;
+                    return origin_status_ref(legacy_intermediate_records.at(identifier));
```

Comment:
At one point I got a "reference to temporary" warning on these lines, but I think changing the return type of `origin_status_ref` to `const` solved it?!